pkg/storage/storage.go

```diff
@@ -66,9 +66,43 @@ func IPManagement(mode int, ipamConf types.IPAMConfig, containerID string) (net.
 	// Check our connectivity first
 	if err := ipam.Status(ctx); err != nil {
 		logging.Errorf("IPAM connectivity error: %v", err)
-		return newip, err
+		return newip, gateway, err
+	}
+
+	for _, ipRange := range ipamConf.Ranges {
+
+		logging.Debugf("try ipRange:%+v", ipRange)
+		newip, err = IPManagementRange(ctx, mode, ipRange, ipam, containerID)
+		logging.Debugf("result allocate in range %+v IP:%+v err:%v", ipRange, newip, err)
+		if err == nil {
+			gateway = ipRange.Gateway
+			break
+
+		} else {
+			switch mode {
+			case 0:
```

Comment:
can you use the constants `types.Allocate` and `types.Deallocate` instead of 0/1?

https://github.com/dougbtv/whereabouts/blob/master/pkg/types/types.go

Reply:
fixed